packages/astro/src/core/routing/rewrite.ts

```diff
@@ -65,7 +65,7 @@ export function findRouteToRewrite({
 		return {
 			routeData: foundRoute,
 			newUrl,
-			pathname,
+			pathname: decodeURI(pathname),
```

Comment:
Nit: since we already do the decoding at line 58, let's do it once: 

```js
let decodedPathname = decodeURI(pathname)
```

And let's use this variable at line 58 and line 68